src/Chainweb/Pact/Backend/Types.hs

```diff
@@ -327,3 +328,7 @@ instance Show PactServiceException where
              ]
 
 instance Exception PactServiceException
+
+data WithCheckpointerResult a
+    = Discard !a
+    | Save BlockHeader !a
```

Comment:
Moved from `PactService.hs` - we had a backend datatype declared in the service code.